application-tour-ui/src/main/resources/TourCode/TourJS.xml

```diff
@@ -249,16 +240,100 @@ require(['jquery', 'xwiki-meta'], function ($, xm) {
+      })
+      .on('focusout', function (event) {
+        if (event.relatedTarget !== null &amp;&amp; !event.currentTarget.parentNode.contains(event.relatedTarget)) {
+          $(nextButtonSelector + ', ' + endButtonSelector).focus();
+          event.stopImmediatePropagation();
+          event.preventDefault();
+        }
+      });
+
+    $(endButtonSelector)
+      .on('click', function (event) {
+        tour.end();
+        event.preventDefault();
+      })
+      .on('keypress', function (event) {
+        // Activate the anchor/button when space or enter are pressed down
+        if (event.which === 32 || event.which === 13) {
+          tour.end();
+        }
+        event.stopImmediatePropagation();
+        event.preventDefault();
+      });
+
+    $(prevButtonSelector)
+      .on('click', function (event) {
+        tour.prev();
+        event.preventDefault();
+      })
+      .on('keypress', function (event) {
+        // Activate the anchor/button when space or enter are pressed down
+        if (event.which === 32 || event.which === 13) {
+          tour.prev();
+        }
+        event.stopImmediatePropagation();
+        event.preventDefault();
+      })
+      .on('focusout', function (event) {
+        if (event.relatedTarget !== null &amp;&amp; !event.currentTarget.parentNode.parentNode.parentNode.contains(event.relatedTarget)) {
+          $(nextButtonSelector + ', ' + endButtonSelector).focus();
+          event.stopImmediatePropagation();
+          event.preventDefault();
+        }
+      });
+
+    $(nextButtonSelector)
+      .on('click', function (event) {
+        tour.next();
+        event.preventDefault();
+      })
+      .on('keypress', function (event) {
+        // Activate the anchor/button when space or enter are pressed down
+        if (event.which === 32 || event.which === 13) {
+          tour.next();
+        }
+        event.stopImmediatePropagation();
+        event.preventDefault();
+      });
+
+    // Wrap around in tab order when the popover is opened, and focus the 'Next' button when the tour step is opened.
+    $(nextButtonSelector + ', ' + endButtonSelector)
+      .on('focusout', function (event) {
+        if (event.relatedTarget !== null &amp;&amp; !event.currentTarget.parentNode.parentNode.parentNode.contains(event.relatedTarget)) {
+          $(closeButtonSelector).focus();
+          event.stopImmediatePropagation();
+          event.preventDefault();
+        }
+      })
+      .focus();
 
-      $(closeButtonSelector).css('position','absolute');
-      $(closeButtonSelector).css('top','3px');
-      $(closeButtonSelector).css('right','5px');
-      // Avoid having the close button on top of the title
-      $('.tour .popover-title').css('padding-right',  $(closeButtonSelector).outerWidth() + 10 + 'px');
+    $(closeButtonSelector).css('position','absolute');
+    $(closeButtonSelector).css('top','3px');
+    $(closeButtonSelector).css('right','5px');
```

Comment:
Can't you use some actual CSS for this?

Reply:
This code was here before the PR. However before this PR there was no style extension object. We can easily add this CSS to the style sheet.
Addressed in https://github.com/xwiki-contrib/application-tour/pull/26/commits/d633e36bdbe4db0ebd8bcf3194da6c5dbec8f590.